pkg/state/snapshot_generator.go

```diff
@@ -791,6 +867,18 @@ func (sg *snapshotGenerator) generateInvokeSnapshot(
 		snapshot = append(snapshot, &assetBalancesSnapshot[i])
 	}
 
+	if sg.IsFullNodeMode && scriptEstimation != nil && scriptRecipient != nil {
+		if scriptEstimation.isPresent() {
```

Comment:
You don't have to check `scriptEstimation != nil`. It's equivalent to the `scriptEstimation.isPresent` implementation.
In which case `scriptRecipient` can be `nil`? `InvokeExpression` transaction?